### ISSUE TITLE
Built-In errors with ShaderInclude files

### ISSUE THREAD

**Issue by clayjohn** [godotengine/godot#63337]
Its working well so far from my testing. I am noticing a few issues though. 

1. When editing a processor function (vertex, fragment etc.) in a ShaderInclude file, there is no highlighting or autocompletion for built-ins. 
2. Stemming from "1." the shader editor reports a compile error if you edit a processor function from a ShaderInclude, yet it actually compiles and runs the shader just fine. 
![Screenshot from 2022-06-29 09-02-28](https://user-images.githubusercontent.com/16521339/176483241-2e9b4669-891d-46b7-9535-404f94a20b95.png)
![Screenshot from 2022-06-29 09-04-48](https://user-images.githubusercontent.com/16521339/176483345-0a9c0866-eb2d-48b3-a25c-777a2fb1a7f6.png)

_Originally posted by @clayjohn in https://github.com/godotengine/godot/issues/62513#issuecomment-1170188124_

**Comment by Chaosus** [comment]
I think the correct behavior is to prevent user to use built-ins within shaderinc - because user can include that shaderinc to the shaders of different types where different built-ins are exist and it would be a mess to support them.

**Comment by clayjohn** [comment]
> I think the correct behavior is to prevent user to use built-ins within shaderinc - because user can include that shaderinc to the shaders of different types where different built-ins are exist and it would be a mess to support them.

I disagree. We need to support buiilt-ins in ShaderInc files so that users can share processor functions between shaders. Right now the feature itself works. The only problem here is that the editor is falsely reporting an error

**Comment by Chaosus** [comment]
> I disagree. We need to support buiilt-ins in ShaderInc files so that users can share processor functions between shaders. Right now the feature itself works. The only problem here is that the editor is falsely reporting an error

Alright, if you think that's okay - I'm not against, just pointing that it may lead to a bit of confusion for the end-user.
